### DISCUSSION TITLE
what is the reference of rfsrc.fast？

### DISCUSSION THREAD

**Discussion by A-Pai** [kogalur/randomForestSRC#146] [Q&A] [answered]
what is the reference of rfsrc.fast？
![image](https://user-images.githubusercontent.com/5112397/147868170-be645188-784e-481f-9253-5c904ca05c6e.png)
I want to figure out its mechanism

**Comment by ishwaran** [comment, marked as answer]
The subsampling rate is generally much smaller than the .632 value used by` rfsrc`.  For` rfsrc.fast`, it is specified by `sampsize` which is 

`sampsize = function(x){min(x * .632, max(150, x ^ (3/4)))}`

This can be seen by consulting the help file https://luminwin.github.io/randomForestSRC/reference/rfsrc.fast.html

Note that by default all rates are sampling without replacement

**Reply by A-Pai** [reply]
what I want to know is why “rfsrc.fast” is fast by subsampling

**Comment by ishwaran** [comment]
Subsampling without replacement means we sample the data randomly and extract a smaller data set of a predetermined size.  So if the sampling rate is small, which is what `rfsrc.fast` does, this means we use a smaller data set each time we grow a tree.  Therefore, each tree in the forest will be constructed much more rapidly.
